pkg/cli/newapp/newapp.go

```diff
@@ -270,6 +270,7 @@ func NewCmdNewApplication(name, baseName string, f kcmdutil.Factory, streams gen
 	o.PrintFlags.AddFlags(cmd)
 
 	cmd.Flags().BoolVar(&o.Config.AsTestDeployment, "as-test", o.Config.AsTestDeployment, "If true create this application as a test deployment, which validates that the deployment succeeds and then scales down.")
+	cmd.Flags().BoolVar(&o.Config.DeploymentConfig, "as-deployment-config", o.Config.DeploymentConfig, "If true create this application as a deployment config, which allows for hooks and custom strategies.")
```

Comment:
Echoing and expanding some of the discussion with @soltysh @smarterclayton and @bparees on slack:

Instead of having a boolean flag here, we should use a string. This gives us flexibility in terms of how we roll out the new behavior.

Ex - change this to `--deploy-with`, with allowed values `deployment` and `deploymentConfig`. We can default to `deploymentConfig` when we introduce the flag, then later switch the default to `deployment`.

Reply:
Why? From what I remember our discussions we've talked about flag. Since this is meant to have only true/false values I don't see any reason why we'd need a string flag, which is also harder to understand for users.

Reply:
My impression was that we were going to keep `DeploymentConfig` as the default initially, but switch to `Deployment` in a future release (4.6?). If we just want to make a clean break for 4.5 with the appropriate release note, I'm OK.

Reply:
Yes, the release note is here: https://github.com/openshift/openshift-docs/issues/19847#issuecomment-606540724